File: Cite.EvalIt.WebApp/src/user-service/core/query/user.lookup.ts
import { Lookup } from '@common/model/lookup';
import { Guid } from '@common/types/guid';
import { IsActive } from '@user-service/core/enum/is-active.enum';
import { UserType } from '@user-service/core/enum/user-type.enum';

export interface UserFilter {
	ids: Guid[];
	excludedIds: Guid[];
	type: UserType[];
	like: string;
	isActive: IsActive[];
}

export class UserLookup extends Lookup implements UserFilter {
	ids: Guid[];
	excludedIds: Guid[];
	type: UserType[];
	like: string;
	isActive: IsActive[];

	constructor() {
		super();
	}
}

File: Cite.EvalIt/Query/UserLookup.cs
using Cite.EvalIt.Common;
using Cite.Tools.Data.Query;
using System;
using System.Collections.Generic;
using System.Text;

namespace Cite.EvalIt.Query
{
	public class UserLookup : Lookup
	{
		public string Like { get; set; }
		public List<Guid> Ids { get; set; }
		public List<Guid> ExcludedIds { get; set; }
		public List<Guid> TagIds { get; set; }
		public List<IsActive> IsActive { get; set; }
		public Guid? ReferenceUserId { get; set; }
		public List<bool> IsNetworkCandidate { get; set; }

		//public UserQuery Enrich(QueryFactory factory)
		//{
		//	UserQuery query = factory.Query<UserQuery>();

		//	if (this.Ids != null) query.Ids(this.Ids);
		//	if (this.ExcludedIds != null) query.ExcludedIds(this.ExcludedIds);
		//	if (this.IsActive != null) query.IsActive(this.IsActive);

		//	this.EnrichCommon(query);

		//	return query;
		//}
	}
}
